### PR TITLE
[languagedetection] Fix panic after detecting JRuby process

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

After detecting a JRuby process, the `DetectLanguage()` function `break`s out of its primary for loop early, causing later processes to not go through the language detection. This means that it returns a `nil` language instead of `Unknown`, which causes a segfault when dereferenced.
 

### Motivation

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4da8bea]

goroutine 342 [running]:
github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta.(*WorkloadMetaExtractor).Extract(0xc000b519a0, 0xc001e8eb70)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta/extractor.go:139 +0x90a
github.com/DataDog/datadog-agent/pkg/process/checks.(*ProcessCheck).run(0xc00142d200, 0x7ff2b4fb, 0x0)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/checks/process.go:255 +0x411
github.com/DataDog/datadog-agent/pkg/process/checks.(*ProcessCheck).Run(0x0?, 0xc000ea3c30, 0xc00141ec90)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/checks/process.go:354 +0xe5
github.com/DataDog/datadog-agent/pkg/process/runner.(*CheckRunner).runCheckWithRealTime(0xc0008c6b40, {0x6515c90, 0xc00142d200}, 0xc00141ec90)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/runner/runner.go:182 +0xc6
github.com/DataDog/datadog-agent/pkg/process/runner.(*CheckRunner).runnerForCheck.func2({0x1, 0x0})
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/runner/runner.go:350 +0x65
github.com/DataDog/datadog-agent/pkg/process/checks.(*runnerWithRealTime).run(0xc0003cc1e0)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/checks/runner.go:59 +0xa8
github.com/DataDog/datadog-agent/pkg/process/runner.(*CheckRunner).Run.func1()
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/runner/runner.go:287 +0x5c
created by github.com/DataDog/datadog-agent/pkg/process/runner.(*CheckRunner).Run
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/process/runner/runner.go:285 +0x3c9
```

### Describe how to test/QA your changes

The process agent should not panic when language detection is enabled on a host with a JRuby process running.

To enable language detection, set the following in datadog.yaml:
```yaml
language_detection:
    enabled: true
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by wiyu — APPROVED** [review]

File: releasenotes/notes/language-detection-panic-b3a5ed023a3332ea.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fixes a panic in the process agent when language detection is enabled on a
```
Comment (buraizu) [inline releasenotes/notes/language-detection-panic-b3a5ed023a3332ea.yaml:11]:
```suggestion
    Fixes a panic in the Process Agent when language detection is enabled on a
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor update

File: releasenotes/notes/language-detection-panic-b3a5ed023a3332ea.yaml
Comment (guyarb) [inline releasenotes/notes/language-detection-panic-b3a5ed023a3332ea.yaml:1]: Not sure we want to put it as a release note
the language detection module is still internal module, it does not affect customers

Comment (robertjli) [reply releasenotes/notes/language-detection-panic-b3a5ed023a3332ea.yaml:1]: Removed, thank you

**Pull request merged**